en/04_Vertex_buffers/00_Vertex_input_description.adoc

```diff
@@ -159,31 +142,29 @@ A bit confusingly, the formats are specified using the same enumeration as color
 The following shader types and formats are commonly used together:
 
 * `float`: `VK_FORMAT_R32_SFLOAT`
-* `vec2`: `VK_FORMAT_R32G32_SFLOAT`
-* `vec3`: `VK_FORMAT_R32G32B32_SFLOAT`
-* `vec4`: `VK_FORMAT_R32G32B32A32_SFLOAT`
+* `float2`: `VK_FORMAT_R32G32_SFLOAT`
+* `float3`: `VK_FORMAT_R32G32B32_SFLOAT`
+* `float4`: `VK_FORMAT_R32G32B32A32_SFLOAT`
 
 As you can see, you should use the format where the amount of color channels matches the number of components in the shader data type.
 It is allowed to use more channels than the number of components in the shader, but they will be silently discarded.
 If the number of channels is lower than the number of components, then the BGA components will use default values of `(0, 0, 1)`.
 The color type (`SFLOAT`, `UINT`, `SINT`) and bit width should also match the type of the shader input.
 See the following examples:
 
-* `ivec2`: `VK_FORMAT_R32G32_SINT`, a 2-component vector of 32-bit signed integers
-* `uvec4`: `VK_FORMAT_R32G32B32A32_UINT`, a 4-component vector of 32-bit unsigned integers
+* `int2`: `VK_FORMAT_R32G32_SINT`, a 2-component vector of 32-bit signed
+integers
+* `uint4`: `VK_FORMAT_R32G32B32A32_UINT`, a 4-component vector of 32-bit
+unsigned integers
 * `double`: `VK_FORMAT_R64_SFLOAT`, a double-precision (64-bit) float
 
-The `format` parameter implicitly defines the byte size of attribute data and the `offset` parameter specifies the number of bytes since the start of the per-vertex data to read from.
+The `format` parameter implicitly defines the byte size of attribute data and the `offset` parameter has specified the number of bytes since the start of the per-vertex data to read from.
 The binding is loading one `Vertex` at a time and the position attribute (`pos`) is at an offset of `0` bytes from the beginning of this struct.
-This is automatically calculated using the `offsetof` macro.
-
-[,c++]
-----
-attributeDescriptions[1].binding = 0;
-attributeDescriptions[1].location = 1;
-attributeDescriptions[1].format = VK_FORMAT_R32G32B32_SFLOAT;
-attributeDescriptions[1].offset = offsetof(Vertex, color);
-----
+This is automatically calculated using the `offsetof` macro.  However, it's
```

Comment:
Not sure if we should keep that paragraph. Feels odd to use `offsetof` and yet warn people to maybe not use it.